src/components/BookmarkIcon/component.tsx

```diff
@@ -0,0 +1,31 @@
+import type React from "react";
+
+import { Bookmark } from "@mui/icons-material";
+import { Button } from "@mui/material";
+
+import type { BookmarkIconProps } from "./types";
+
+const BookmarkIcon: React.FC<BookmarkIconProps> = () => {
+  return (
+    <Button
+      variant="outlined"
+      size="small"
+      startIcon={
+        <Bookmark
+          sx={{
+            fill: "transparent",
+            stroke: "#FFF",
+            strokeWidth: "2px",
+          }}
+        />
+      }
+      sx={{
+        "& .MuiButton-startIcon": { margin: 0 },
+        "&:hover .MuiButton-startIcon svg": { stroke: "#10141E" },
+        "&:focus .MuiButton-startIcon svg": { fill: "#FFFFFF" },
```

Comment:
For active button you should use prop isActive, and change styles, setup current background color of icon instead pseudo-class focus, that will depend on prop isActive.